router/customdestinationmanager/customdestinationmanager.go

```diff
@@ -133,13 +133,25 @@ func (customManager *CustomManagerT) send(jsonData json.RawMessage, client inter
 	case KV:
 		var err error
 		kvManager, _ := client.(kvstoremanager.KVStoreManager)
-		// if the event supports HSET operation then use HSET
-		if kvstoremanager.IsHSETCompatibleEvent(jsonData) {
-			hash, key, value := kvstoremanager.ExtractHashKeyValueFromEvent(jsonData)
-			err = kvManager.HSet(hash, key, value)
+		// if the event type is record
+		if kvstoremanager.IsRecordCompatibleEvent(jsonData) {
```

Comment:
We should not bring the context of `record` here. Just check if the payload is compatible with delete.